app/core/exceptions/course.py

```diff
@@ -0,0 +1,16 @@
+from .base import CustomException
+
+class MultipleCoursesExistException(CustomException):
+    code = 500
+    error_code = "COURSE__MULTIPLE_COURSES_EXIST"
+    message = "database misconfigured, multiple courses exist in course table"
+
+class NoCourseExistsException(CustomException):
+    code = 500
+    error_code = "COURSE__NO_COURSE_EXISTS"
+    message = "database misconfigured, no course exists in course table"
+
+class CourseAlreadyExistsException(CustomException):
+    code = 403
```

Comment:
minor, but I dont think 403 is the one we need. 409 might be more appropriate, or 400 -

```400 Bad Request - when the server will not process a request because it's obvious client fault
409 Conflict - if the server will not process a request, but the reason for that is not the client's fault```

Reply:
ah I just saw you used 409 for `UserAlreadyExists`, we could be consistent if we used 409 here too

Reply:
Yeah this should not be a 403. I'm honestly in favor of 400 here and changing the other one as well. 409 is most appropriate, although I think it's generally better to avoid the specialized codes since they are very rarely actually used and harder to recognize.

Reply:
I personally think there's some utility to using error codes like 409. I think there's some error codes that are really specific and that we won't get any use out of in our applications (e.g. 410 "Gone", which is to indicate that the resource is no longer present and won't be coming back, and that clients should purge it from their caches). But others, like 409 are useful in this scenario because they keep us from diluting 400. If we just use 200 for every success and 400 for every client failure and 500 for every server failure we're diluting the pool of what those statuses mean. I just feel like being more specific is better when it makes sense so that clients don't have to infer what the error is, and I think 409 makes sense here.

Reply:
We can use 409. Although the benefit of differentiation here is more or less lost since we assign unique codes to all our errors anyways.